### ISSUE TITLE
[Framework] trusted_proxies: private_ranges

### ISSUE THREAD

**Issue by dkarlovi** [symfony/symfony#51826]
### Description

Currently, if your app is behind a reverse proxy, you need to enable trusted proxies to be sure Symfony will understand it's using HTTPS.

Typically it looks like this:

```yaml
framework:
    # localhost, Docker, Kubernetes
    trusted_proxies: 127.0.0.1, 172.16.0.0/12, 10.0.0.0/8
```

[Caddy solves the same issue by having a dedicated alias called `private_ranges`](https://caddyserver.com/docs/caddyfile/options#trusted-proxies):

> As a shortcut, private_ranges can be used to match all private IPv4 and IPv6 ranges. It's the same as specifying all of these ranges: 192.168.0.0/16 172.16.0.0/12 10.0.0.0/8 127.0.0.1/8 fd00::/8 ::1

Proposal is to introduce the same alias for these specific values to allow easier configuration in this circumstance.

### Example

```yaml
framework:
    trusted_proxies: private_ranges
```

**Comment by magikid** [comment]
I like this idea.

**Comment by xabbuh** [comment]
I also like this idea and implemented it in #52924.
